book/src/proposals/booting-from-snapshot.md

```diff
@@ -0,0 +1,262 @@
+# Booting From a Snapshot
+
+## Problem
+
+A validator booting from a snapshot should confirm that they are caught up to
+the correct cluster before voting. There are two major risks if they do not:
+
+### Wrong Cluster
+
+The validator may be fed a snapshot and gossip info for a different (though not
+necessarily malicious) cluster. If that cluster is at slot 100,000 while the
+cluster the validator intended to join is at 10,000, if the validator votes
+once on the wrong cluster, it must not vote on the correct cluster for at least
+90,000 slots to avoid being slashed. This is because slashing is based on slot
+numbers, so when the validator makes a vote `V` on the wrong cluster's slot
+100,000, it is locked out of voting on any slot < `100,000 + lockout(V)` on the
+correct cluster.
+
+### Voting Before Catching Up
```

Comment:
Looks like the risk here is "accidentally slashed for double-voting", but that's not obvious to me, because if the voting algorithm is deterministic, they will vote for the same fork at the same height, which is not a slashing condition.  Is there something missing here? Is there some nondeterminism that makes this a realistic concern? Or just being risk adverse?

Reply:
I don't understand the question here. In the case of two clusters, the validator is starting with two different states, one for the wrong cluster and one for the right cluster. The voting algorithm will not produce votes for the same forks/heights because they don't exist in both states. This will cause a slashing condition in one direction, the only question is if it's in the direction you care about.

Reply:
I don't see how catching up is relevant. If you vote early on the correct cluster, it's no problem. If you vote anywhere on the wrong cluster, caught up or now, you've got a problem.

Reply:
Do the new clarifications help?